_overviews/overview/concurrency.md

```diff
@@ -0,0 +1,267 @@
+in the 'yield': 806
+in the Success case: 806
+result = 6
+````
+
+As that output shows, the futures are created very rapidly, and in just two milliseconds the print statement right before the `sleep(3000)` statement at the end of the method is reached. All of that code is run on the JVM’s main thread. Then, at 806 ms, the three futures complete and the code in the `yield` block is run. Then the code immediately goes to the `Success` case in the `onComplete` method.
+
+The 806 ms output is a key to seeing that the three futures are run in parallel. If they were run sequentially, the total time would be about 1,400 ms — the sum of the sleep times of the three futures. But because they’re run in parallel, the total time is just slightly longer than the longest-running future: `f1`, which is 800 ms.
+
+
+### A method that returns a future
+
+So far you’ve seen how to pass a single-threaded algorithm into a `Future` constructor. You can use the same technique to create a method that returns a `Future`:
+
+```scala
+// simulate a slow-running method
+def slowlyDouble(x: Int, delay: Long): Future[Int] = Future {
+  sleep(delay)
+  x * 2
+}
+```
+
+As with the previous examples, just assign the result of the method call to a new variable. Then when you check the result right away you’ll see that it’s not completed, but after the delay time the future will have a result:
+
+````
+scala> val f = slowlyDouble(2, 5_000L)                                       val f: concurrent.Future[Int] = Future(<not completed>)
+
+scala> f
+val res0: concurrent.Future[Int] = Future(<not completed>)
+
+scala> f
+val res1: concurrent.Future[Int] = Future(Success(4))
+````
+
+
+
+## Key points about futures
+
+Hopefully those examples give you an idea of how Scala futures work. To summarize, a few key points about futures are:
+
+- You construct futures to run tasks off of the main thread
+- Futures are intended for one-shot, potentially long-running concurrent tasks that *eventually* return a value; they create a temporary pocket of concurrency
+- A future starts running as soon as you construct it
+- A benefit of futures over threads is that they work with `for` expressions, and come with a variety of callback methods that simplify the process of working with concurrent threads
+- When you work with futures you don’t have to concern yourself with the low-level details of thread management
+- You handle the result of a future with callback methods like `onComplete` and `andThen`, or transformation methods like `filter`, `map`, etc.
+- The value inside a `Future` is always an instance of one of the `Try` types: `Success` or `Failure`
+- If you’re using multiple futures to yield a single result, combine them in a `for` expression
+
+Also, as you saw with the `import` statements in these examples, the Scala `Future` depends on an `ExecutionContext`. This relationship, and many more details about futures, are explained in the Reference documentation.
```

Comment:
Maybe say one sentence about the execution context before referring to the reference docs?